lib/Test/Nginx/Socket/Lua/Dgram.pm

```diff
@@ -0,0 +1,445 @@
+package Test::Nginx::Socket::Lua::Dgram;
+
+use v5.10.1;
+use Test::Nginx::Socket::Lua -Base;
+use Test::Nginx::Util qw( $ServerPort $ServerAddr );
+
+sub get_best_long_bracket_level ($);
+sub quote_as_lua_str ($);
+sub gen_data_file ($);
+
+my $port = $ENV{TEST_NGINX_DGRAM_PORT} // ($ServerPort + 1);
+my $counter = 0;
+
+add_block_preprocessor(sub {
+    my ($block) = @_;
+
+    my $name = $block->name;
+
+    my $dgram_config = $block->dgram_config;
+    my $dgram_server_config = $block->dgram_server_config;
+    my $dgram_server_config2 = $block->dgram_server_config2;
+    my $dgram_server_config3 = $block->dgram_server_config3;
+    my $dgram_req = $block->dgram_request;
```

Comment:
I think we should make `dgram_request` mandatory since otherwise the server won't get triggered at all (we do not have the notion of connections in the UDP world).